saleor/tax/utils.py

```diff
@@ -248,3 +253,82 @@ def get_shipping_tax_class_kwargs_for_order(tax_class: Optional["TaxClass"]):
         "shipping_tax_class_private_metadata": tax_class.private_metadata,
         "shipping_tax_class_metadata": tax_class.metadata,
     }
+
+
+def validate_tax_data(
+    tax_data: Optional[TaxData],
+    lines: Iterable,
+    allow_empty_tax_data: bool = False,
+):
+    if tax_data is None and not allow_empty_tax_data:
+        raise TaxDataError(TaxDataErrorMessage.EMPTY)
+
+    if check_negative_values_in_tax_data(tax_data):
+        logger.warning(TaxDataErrorMessage.NEGATIVE_VALUE)
+        raise TaxDataError(TaxDataErrorMessage.NEGATIVE_VALUE)
+
+    if check_line_number_in_tax_data(tax_data, lines):
+        logger.warning(TaxDataErrorMessage.LINE_NUMBER)
+        raise TaxDataError(TaxDataErrorMessage.LINE_NUMBER)
+
+    if check_overflows_in_tax_data(tax_data):
+        logger.warning(TaxDataErrorMessage.OVERFLOW)
+        raise TaxDataError(TaxDataErrorMessage.OVERFLOW)
+
+
+def check_negative_values_in_tax_data(tax_data: Optional[TaxData]) -> bool:
+    """Check if tax data contains negative values."""
+    if not tax_data:
+        return False
+
+    if (
+        tax_data.shipping_price_gross_amount < 0
+        or tax_data.shipping_price_net_amount < 0
+        or tax_data.shipping_tax_rate < 0
+    ):
+        return True
+
+    for line in tax_data.lines:
+        if (
+            line.total_gross_amount < 0
+            or line.total_net_amount < 0
+            or line.tax_rate < 0
+        ):
+            return True
+
+    return False
+
+
+def check_line_number_in_tax_data(tax_data: Optional[TaxData], lines: Iterable) -> bool:
+    """Check if tax data contains same line number as input data."""
+    if not tax_data:
+        return False
+
+    if len(tax_data.lines) != len(list(lines)):
+        return True
+
+    return False
+
+
+def check_overflows_in_tax_data(tax_data: Optional[TaxData]) -> bool:
+    """Check if tax rates exceed 100% and line prices are lower than a billion."""
+    if not tax_data:
+        return False
+
+    max_price = 999999999
```

Comment:
Cosmetic change: Maybe we should have a constant at the top of the file for this value (optionally with a comment explaining why billion).